src/components/graph/layout.ts

```diff
@@ -261,22 +251,23 @@ function compressTreePlacements(nodes: CurrentTreeNode[], placements: PlacementG
         // to the maximum column placement values (per row) of the nodes on the left.
         // The resulting space we find represents how much we can shift the current column to the left.
 
-        const currentSubTreeSize = 1 + countNodes(nodes, currentNodeId as UUID);
-        const indexOfCurrentNode = nodes.findIndex((n) => n.id === currentNodeId);
-        const nodesOfTheCurrentBranch = nodes.slice(indexOfCurrentNode, indexOfCurrentNode + currentSubTreeSize);
-        const currentBranchMinimumColumnByRow = getMinimumColumnByRows(nodesOfTheCurrentBranch, placements);
+        const { index: currentNodeIndex } = nodeMap.get(currentNodeId)!;
```

Comment:
Why not 
`const currentNodeIndex = nodeMap.get(currentNodeId)!.index;`
This is clearer.